BeanHub.API/src/main/java/com/bean/api/services/RatingService.java

```diff
@@ -46,26 +42,21 @@ public Rating getRatingById(Long ratingId) {
         return entityManager.find(Rating.class, ratingId);
     }
 
-    public List<Recipe> getSortedFeed(int userId, String sort) {
-
+    public List<Recipe> getSortedFeed(int userId, RatingEnums.SortType sortType) {
         String jpql = "SELECT r.recipe.recipeId FROM Rating r WHERE r.user.userId = :userId";
         TypedQuery<Long> query = entityManager.createQuery(jpql, Long.class);
         query.setParameter("userId", userId);
         List<Long> recipeIds = query.getResultList();
 
         Map<Long, Double> averageRatings = getAverageRatings(recipeIds);
 
-        switch (sort) {
-            case "1":
-                return getSortedRecipesByDateAdded(recipeIds, Comparator.nullsLast(Comparator.reverseOrder()));
-            case "2":
-                return getSortedRecipesByDateAdded(recipeIds, Comparator.nullsLast(Comparator.naturalOrder()));
-            case "3":
-                return getSortedRecipesByAverageRating(recipeIds, averageRatings, sort);
-            case "4":
-                return getSortedRecipesByAverageRating(recipeIds, averageRatings, sort);
+        switch (sortType) {
```

Comment:
Do these cases not need to be refered to by saying EnumName.VALUE?
if not this is fine

Reply:
As far as I understand the value can be accessed directly without EnumName. From the tests that I've done the results are as expected